app/controllers/api/v1/sources_controller.rb

```diff
@@ -23,19 +23,24 @@ def check_availability
         source = Source.find(params[:source_id])
         topic  = "platform.topological-inventory.operations-#{source.source_type.name}"
 
-        logger.debug("publishing message to #{topic}...")
-        messaging_client = Sources::Api::Messaging.client(topic)
-        messaging_client.publish_topic(
-          :service => topic,
-          :event   => "Source.availability_check",
-          :payload => {
-            :params => {
-              :source_id       => source.id.to_s,
-              :external_tenant => source.tenant.external_tenant
+        if Sources::Api::Messaging.topics.include?(topic)
+          logger.debug("publishing message to #{topic}...")
+
+          Sources::Api::Messaging.client.publish_topic(
+            :service => topic,
+            :event   => "Source.availability_check",
+            :payload => {
+              :params => {
+                :source_id       => source.id.to_s,
+                :external_tenant => source.tenant.external_tenant
+              }
             }
-          }
-        )
-        logger.debug("publishing message to #{topic}...Complete")
+          )
+
+          logger.debug("publishing message to #{topic}...Complete")
+        else
+          logger.debug("not publishing message to non-existing topic: #{topic}")
```

Comment:
Ideally this shouldn't happen ... can we make this an error or warn?

Reply:
So the way the availability checker works right now this actually happens rather often, and not just for source-types that we don't have an operations worker for yet.

E.g. There are the cloudforms source-types like ovirt, vsphere, openstack where I don't think we'll ever have an operations worker.

Reply:
I think the way to properly fix this, is to turn this current check into the topological-inventory application check.  So sources that aren't enabled for topology don't try to write to this topic.  Then topology is playing at the same level as all the other apps on the platform.

Reply:
Yeah, I was thinking this was happening pretty often, but we log debug by default right now anyway so I was thinking making this an error would make the failing topics easier to find.

Reply:
:+1: done